tests/scripts/link/script.cli

```diff
@@ -0,0 +1,234 @@
+# Functional tests for project linking
+# Key:
+#   - # <comment>
+#   - $ <command>
+#   - ? <grok pattern>
+#   - ! <literal>
+#   - repeat { ... }
+#   - option { ... }
+#   - ignore { ... }
+#   - choice { ... }
+#   - unordered { ... }
+
+#pattern VERSION \d+\.\d+(?:\.\d+)?(?:-(?:alpha|beta|rc|dev)\.\d+)?\+(?:[a-f0-9]{7}|local)
+#test %{VERSION} "Gel CLI 7.2.0-dev.1306+7d4a485"
+ignore {
+    ? Newer version of gel tool exists %{VERSION} \(current %{VERSION}\). To upgrade run `gel cli upgrade`
+    ? WARNING %{GREEDYDATA} postgres: %{GREEDYDATA}
+    ? CRITICAL %{GREEDYDATA} postgres: %{GREEDYDATA}
+}
+
+$ gel instance destroy -I inst1 --force
+%ALLOW_FAILURE
+*
+
+$ gel instance destroy -I project2 --force
+%ALLOW_FAILURE
+*
+
+$ gel --version
+? Gel CLI %{VERSION}
+
+$ gel server list-versions
+? ┌%{DATA}┬%{DATA}┬%{DATA}┐
+? │%{DATA}│%{DATA}│%{DATA}│
+repeat {
+    choice {
+        ? │ stable %{DATA}
+        ? │ nightly %{DATA}
+        ? │ testing %{DATA}
+    }
+}
+? └%{DATA}┴%{DATA}┴%{DATA}┘
+
+$ gel instance create inst1
+choice {
+    sequence {
+        ? Downloading package...
+        ? %{DATA}
+        ? Successfully installed %{VERSION}
+    }
+    ? Version %{VERSION} is already downloaded
+}
+! Initializing Gel instance...
+choice {
+    ! Gel will not start on next login. Trying to start database in the background...
+    repeat {
+        ? Created symlink %{GREEDYDATA}
+    }
+}
+! Instance inst1 is up and running.
+! To connect to the instance run:
+!   gel -I inst1
+
+$ gel project info --instance-name --project-dir=tests/proj/project1
+%EXIT 1
+! gel error: Project is not initialized. Run `gel project init`.
```

Comment:
What would you think about also placing contents of all needed files into these scripts?
Like have a header that says:
```
file "tests/proj/project1/gel.toml" {
[project]
schema-dir = "dbschema"
}
``` 
That would make these files more isolated and portable. We would have to make a mechanism that creates a tmp dir, writes the files and removes them after the test (or mock the file system directly).

Reply:
Let me double-check, but heredocs might work for this...